ts/store/reducers/wallet/payment.ts

```diff
@@ -0,0 +1,123 @@
+import { none, Option, some } from "fp-ts/lib/Option";
+import { AmountInEuroCents, RptId } from "italia-ts-commons/lib/pagopa";
+import { createSelector } from "reselect";
+import { EnteBeneficiario } from "../../../../definitions/pagopa-proxy/EnteBeneficiario";
+import { PaymentRequestsGetResponse } from "../../../../definitions/pagopa-proxy/PaymentRequestsGetResponse";
+import { Wallet } from "../../../../definitions/pagopa/Wallet";
+import {
+  PAYMENT_STORE_SELECTED_PAYMENT_METHOD,
+  STORE_INITIAL_AMOUNT,
+  STORE_RPTID_DATA,
+  STORE_VERIFICA_DATA
+} from "../../actions/constants";
+import { Action } from "../../actions/types";
+import { GlobalState } from "../types";
+import { getCardFromId, getCards } from "./cards";
+
+export type PaymentState = Readonly<{
+  rptId: Option<RptId>;
+  verificaResponse: Option<PaymentRequestsGetResponse>;
+  initialAmount: Option<AmountInEuroCents>;
+  selectedPaymentMethod: Option<number>;
+}>;
+
+export const PAYMENT_INITIAL_STATE: PaymentState = {
+  rptId: none,
+  verificaResponse: none,
+  initialAmount: none,
+  selectedPaymentMethod: none
+};
+
+export const getVerificaResponse = (
+  state: GlobalState
+): Option<PaymentRequestsGetResponse> => state.wallet.payment.verificaResponse;
+
+export const getRptId = (state: GlobalState): Option<RptId> =>
+  state.wallet.payment.rptId;
+
+export const getInitialAmount = (
+  state: GlobalState
+): Option<AmountInEuroCents> => state.wallet.payment.initialAmount;
+
+export const getSelectedPaymentMethod = (state: GlobalState): Option<number> =>
+  state.wallet.payment.selectedPaymentMethod;
+
+export const currentAmountSelector = createSelector(
+  getVerificaResponse,
+  (rsp: Option<PaymentRequestsGetResponse>): Option<AmountInEuroCents> => {
+    if (rsp.isNone()) {
+      return none;
+    }
+    // TODO: update proxy to return AmountInEuroCents
+    return some((
+      "0".repeat(10) + `${rsp.value.importoSingoloVersamento}`
+    ).slice(-10) as AmountInEuroCents);
+  }
+);
+
+export const paymentRecipientSelector = createSelector(
+  getVerificaResponse,
+  (rsp: Option<PaymentRequestsGetResponse>): Option<EnteBeneficiario> => {
+    if (rsp.isNone() || rsp.value.enteBeneficiario === undefined) {
```

Comment:
```ts
rsp.mapNullable(_ => _.enteBeneficiario); 
```

https://github.com/gcanti/fp-ts/blob/master/src/Option.ts#L249

almost always when you're checking if an option is empty or not to do something on its value, there's a method that already covers your case